ZelBack/src/services/appsService.js

```diff
@@ -6056,30 +6069,71 @@ async function storeAppRunningMessage(message) {
 
   const db = dbHelper.databaseConnection();
   const database = db.db(config.database.appsglobal.database);
-  const newAppRunningMessage = {
-    name: message.name,
-    hash: message.hash, // hash of application specifics that are running
-    ip: message.ip,
-    broadcastedAt: new Date(message.broadcastedAt),
-    expireAt: new Date(validTill),
-  };
 
-  // indexes over name, hash, ip. Then name + ip and name + ip + broadcastedAt.
-  const queryFind = { name: newAppRunningMessage.name, ip: newAppRunningMessage.ip, broadcastedAt: { $gte: newAppRunningMessage.broadcastedAt } };
-  const projection = { _id: 0 };
-  // we already have the exact same data
-  const result = await dbHelper.findOneInDatabase(database, globalAppsLocations, queryFind, projection);
-  if (result) {
-    // it is already stored
-    return false;
+  if (message.version === 1) {
+    const newAppRunningMessage = {
+      name: message.name,
+      hash: message.hash, // hash of application specifics that are running
+      ip: message.ip,
+      broadcastedAt: new Date(message.broadcastedAt),
+      expireAt: new Date(validTill),
+    };
+
+    // indexes over name, hash, ip. Then name + ip and name + ip + broadcastedAt.
+    const queryFind = { name: newAppRunningMessage.name, ip: newAppRunningMessage.ip, broadcastedAt: { $gte: newAppRunningMessage.broadcastedAt } };
+    const projection = { _id: 0 };
+    // we already have the exact same data
+    const result = await dbHelper.findOneInDatabase(database, globalAppsLocations, queryFind, projection);
+    if (result) {
+      // it is already stored
+      return false;
+    }
+    const queryUpdate = { name: newAppRunningMessage.name, ip: newAppRunningMessage.ip };
+    const update = { $set: newAppRunningMessage };
+    const options = {
+      upsert: true,
+    };
+    await dbHelper.updateOneInDatabase(database, globalAppsLocations, queryUpdate, update, options);
+  } else {
+    let messageNotOk = false;
+    for (let i = 0; i < message.apps.length; i += 1) {
+      const app = message.apps[i];
+      if (typeof app.hash !== 'string' || typeof app.name !== 'string') {
+        return new Error('Invalid Flux App Running v2 message for storing');
```

Comment:
This type check is what should happen earlier (just so we have type validation at one place)